app/controllers/PdfDownloadController.scala

```diff
@@ -39,47 +40,49 @@ class PdfDownloadController @Inject()(appConfig: FrontendAppConfig,
                                      ) extends FrontendController with I18nSupport {
 
   def application(reference: String, token: Option[String]): Action[AnyContent] = identify.async { implicit request =>
-    if(request.eoriNumber.isDefined) {
-      getApplicationPDF(request.eoriNumber.get, reference)
-    } else if(token.isDefined) {
-      pdfService.decodeToken(token.get) match {
-        case Some(eori) => getApplicationPDF(eori, reference)
-        case None => Future.successful(Redirect(controllers.routes.SessionExpiredController.onPageLoad()))
-      }
-    } else {
-      Future.successful(Redirect(controllers.routes.UnauthorisedController.onPageLoad()))
-    }
+    getPdf(reference, token, getApplicationPDF)
   }
 
   def ruling(reference: String, token: Option[String]): Action[AnyContent] = identify.async { implicit request =>
```

Comment:
there were two methods doing exactly the same thing... so I created a generic method used by both